src/groupWith.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
```

Comment:
I struggled with the explanation for this function. I think it's easier to explain as the version of `group` that allows a user-supplied function. How does this sound?
> Takes a list and returns a list of lists where each sublist contains only elements which satisfy the user supplied equality function.